### PR TITLE
CXF-94667-validate-metal: Validate metal connection Status

### PR DESCRIPTION
**Pull request by d-bhola**
PR has changes for all metal examples (metal on aSide or zSide)- 
- Add resource (type: `time_sleep`, name: `wait_dl_connection`) to wait for 2min, until the connection is Provisioned on fabric side
- Add data source (type: `equinix_metal_connection`, name: `NIMF-test`) to fetch the metal-connection id
- Add output (name: `metal_connection_status`) to get the status of metal connection

_Metal to AWS, Metal to Azure, Metal to Port, Cloud Router to Metal, Metal to Service Profile are successfully passing the tests._

### CODE REVIEW

File: examples/cloud-router-2-metal-nimf-connection/outputs.tf [hunk 1]
```diff
@@ -7,3 +7,9 @@ output "metal_connection_id" {
 output "cloud_router_metal_connection_id" {
   value = module.cloud_router_2_metal_connection.primary_connection_id
 }
+output "cloud_router_routing_protocol_id" {
+  value = module.routing_protocols.direct_routing_protocol_id
+}
+output "metal_connection_status" {
+  value = data.equinix_metal_connection.NIMF-test.status
+}
```
Comment (srushti-patl) [inline examples/cloud-router-2-metal-nimf-connection/outputs.tf:15]: nit: Add extra space at the end of the file

File: examples/metal-nimf-2-aws-connection/main.tf [hunk 2]
```diff
@@ -76,3 +76,13 @@ resource "aws_dx_private_virtual_interface" "aws_virtual_interface" {
...
+
+resource "time_sleep" "wait_dl_connection" {
+  depends_on = [module.metal_2_aws_connection]
+  create_duration = "2m"
+}
+
+data "equinix_metal_connection" "NIMF-test" {
+  depends_on = [time_sleep.wait_dl_connection]
+  connection_id     = equinix_metal_connection.metal-connection.id
+}
```
Comment (srushti-patl) [inline examples/metal-nimf-2-aws-connection/main.tf:88]: nit: Add extra space and format it properly

File: examples/metal-nimf-2-aws-connection/outputs.tf [hunk 3]
```diff
@@ -13,3 +13,6 @@ output "aws_dx_gateway_id" {
 output "aws_interface_id" {
   value = aws_dx_private_virtual_interface.aws_virtual_interface.id
 }
+output "metal_connection_status" {
+  value = data.equinix_metal_connection.NIMF-test.status
+}
```
Comment (srushti-patl) [inline examples/metal-nimf-2-aws-connection/outputs.tf:18]: nit: Add extra space

File: examples/metal-nimf-2-azure-connection/main.tf [hunk 4]
```diff
@@ -68,3 +68,12 @@ module "metal_2_azure_connection" {
...
 }
+resource "time_sleep" "wait_dl_connection" {
+  depends_on = [module.metal_2_azure_connection]
+  create_duration = "2m"
+}
+
+data "equinix_metal_connection" "NIMF-test" {
+  depends_on = [time_sleep.wait_dl_connection]
+  connection_id     = equinix_metal_connection.metal-connection.id
+}
```
Comment (srushti-patl) [inline examples/metal-nimf-2-azure-connection/main.tf:79]: nit: Add extra space and format it properly.

File: examples/metal-nimf-2-ibm2-connection/main.tf [hunk 5]
```diff
@@ -71,3 +72,7 @@ resource "ibm_dl_gateway_action" "test_dl_gateway_action" {
   metered         = var.ibm_gateway_metered
   resource_group  = data.ibm_resource_group.rg.id
 }
+
+data "equinix_metal_connection" "NIMF-test" {
+  connection_id     = equinix_metal_connection.metal-connection.id
```
Comment (srushti-patl) [inline examples/metal-nimf-2-ibm2-connection/main.tf:77]: Add depends_on = [time_sleep.wait_dl_connection] line in Metal Connection data source block

File: examples/metal-nimf-2-oracle-connection/main.tf [hunk 6]
```diff
@@ -78,3 +78,12 @@ module "metal_2_oracle_connection" {
...
   zside_fabric_sp_name        = var.zside_fabric_sp_name
 }
+
+resource "time_sleep" "wait_dl_connection" {
+  depends_on = [module.metal_2_oracle_connection]
+  create_duration = "2m"
+}
+
+data "equinix_metal_connection" "NIMF-test" {
+  connection_id     = equinix_metal_connection.metal-connection.id
```
Comment (srushti-patl) [inline examples/metal-nimf-2-oracle-connection/main.tf:88]: Add depends_on = [time_sleep.wait_dl_connection] line in Metal Connection data source block

File: examples/metal-nimf-2-port-connection/main.tf [hunk 7]
```diff
@@ -37,3 +37,11 @@ module "metal_2_port_connection" {
   zside_vlan_tag        = var.zside_vlan_tag
   zside_location        = var.zside_location
 }
+
+resource "time_sleep" "wait_dl_connection" {
```
Comment (srushti-patl) [inline examples/metal-nimf-2-port-connection/main.tf:41]: Add depends_on line in time sleep block

File: examples/metal-nimf-2-port-connection/main.tf [hunk 8]
```diff
@@ -37,3 +37,11 @@ module "metal_2_port_connection" {
...
+  create_duration = "2m"
+}
+
+data "equinix_metal_connection" "NIMF-test" {
+  connection_id     = equinix_metal_connection.metal-connection.id
```
Comment (srushti-patl) [inline examples/metal-nimf-2-port-connection/main.tf:46]: Add depends_on = [time_sleep.wait_dl_connection] line in Metal Connection data source block

File: examples/metal-nimf-2-service-profile-connection/main.tf [hunk 9]
```diff
@@ -38,3 +38,12 @@ module "metal_2_service_profile" {
   zside_location        = var.zside_location
   zside_fabric_sp_name  = var.zside_fabric_sp_name
 }
+
+resource "time_sleep" "wait_connection" {
+  depends_on = [module.metal_2_service_profile]
+  create_duration = "2m"
+}
+
+data "equinix_metal_connection" "NIMF-test" {
```
Comment (srushti-patl) [inline examples/metal-nimf-2-service-profile-connection/main.tf:47]: Add depends_on = [time_sleep.wait_dl_connection] line in Metal Connection data source block

File: examples/metal-nimf-2-service-profile-connection/main.tf [hunk 10]
```diff
@@ -38,3 +38,12 @@ module "metal_2_service_profile" {
...
+  connection_id     = equinix_metal_connection.metal-connection.id
+}
```
Comment (srushti-patl) [inline examples/metal-nimf-2-service-profile-connection/main.tf:49]: nit: Add extra space

File: modules/cloud-router-routing-protocols/main.tf [hunk 11]
```diff
@@ -30,4 +30,10 @@ resource "equinix_fabric_routing_protocol" "bgp" {
     enabled          = var.bgp_enabled_ipv6
     customer_peer_ip = var.bgp_customer_peer_ipv6
   }
+  dynamic "bgp_ipv6" {
+    for_each = var.bgp_customer_peer_ipv6 != "" ? [1] : []
+    content {
+      customer_peer_ip = var.bgp_customer_peer_ipv6
+    }
+  }
```
Comment (srushti-patl) [inline modules/cloud-router-routing-protocols/main.tf:38]: Remove dulplicate bgp_ipv6 block and Add similar dynamic block for direct_ipv6

**Comment by d-bhola** [comment]
Addressed all comments. Thanks!

**Pull request merged**